lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -357,6 +414,23 @@ export default function getRecipeInterface(querier: Querier, config: TypeNormali
             return SessionFunctions.updateAccessTokenPayload(helpers, sessionHandle, newAccessTokenPayload);
         },
 
+        mergeIntoAccessTokenPayload: async function ({
+            sessionHandle,
+            accessTokenPayloadUpdate,
+        }: {
+            sessionHandle: string;
+            accessTokenPayloadUpdate: JSONObject;
+        }) {
+            const sessionInfo = await SessionFunctions.getSessionInformation(helpers, sessionHandle);
+            const updatedPayload = { ...sessionInfo.accessTokenPayload, ...accessTokenPayloadUpdate };
+            for (const key of Object.keys(accessTokenPayloadUpdate)) {
+                if (accessTokenPayloadUpdate[key] === null) {
+                    delete updatedPayload[key];
+                }
+            }
+            return this.updateAccessTokenPayload(sessionHandle, updatedPayload);
+        },
```

Comment:
Sure that we want to do this like this? Or make changes to the core to have an API that does it in a thread safe way?